### PR TITLE
Refactor `testVulnerability` for concurrent execution and time it

### PR DESCRIPTION
**Pull request by lloydlobo**
## What's new

This PR refactors `testVulnerability` function to improve test performance.

- ~~The test function now runs concurrently using `Promise.all` and `async` funtions.~~
- ~~Uses a timer to measure execution time.~~
- With tests out of `src` directory, a dedicated testing framework handles compilation 
  and testing out of the box. [See reference](https://github.com/prayas7102/NodejsSecurify/pull/19#issuecomment-2437017124).
- Documents the new test suite in `CONTRIBUTING.md`.

### CODE REVIEW

**Comment by prayas7102** [comment]
Thanks @lloydlobo will review it shortly.

**Comment by prayas7102** [comment]
@lloydlobo can we add Vulnerability_test.ts outside of src directory? Because it seems only for testing purpose, not for production.

File: tests/Vulnerability/DatasetSample.spec.ts [hunk 1]
```diff
@@ -23,7 +23,6 @@ describe('Dataset Map Validations', () => {
             expect(invalidEntries).toHaveLength(0);
         });
         invalidSamples.forEach((sample, index) => {
-            const vulnLabel = toStringMap[vulnerability];
```
Comment (lloydlobo) [inline tests/Vulnerability/DatasetSample.spec.ts]: Removing this line, avoids shadowing the constant with the same name in ancestor scope:

```ts
 vulnerabilities.forEach((vulnerability) => {
        const vulnLabel = toStringMap[vulnerability]; // <<< This one was being shadowed
        it(`should validate valid dataset for '${vulnLabel}'`, () => {
```

**Comment by prayas7102** [comment]
@lloydlobo can we have a concise documentation for the developers to use this test scripts in contributing.md?

**Review by prayas7102 — APPROVED** [review]

**Pull request merged**